api/konnect/v1alpha1/konnect_cloudgateway_transitgateway_types.go

```diff
@@ -0,0 +1,203 @@
+
+	commonv1alpha1 "github.com/kong/kubernetes-configuration/api/common/v1alpha1"
+)
+
+func init() {
+	SchemeBuilder.Register(&KonnectCloudGatewayTransitGateway{}, &KonnectCloudGatewayTransitGatewayList{})
+}
+
+// KonnectCloudGatewayTransitGateway is the Schema for the Konnect Transit Gateway API.
+//
+// +genclient
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:resource:categories=kong
+// +kubebuilder:object:root=true
+// +kubebuilder:object:generate=true
+// +kubebuilder:subresource:status
+// +kubebuilder:subresource:finalizer
+// +apireference:kgo:include
+// +kubebuilder:printcolumn:name="State",type=string,JSONPath=`.status.state`,description="State of the transit gateway in Konnect"
+// +kubebuilder:validation:XValidation:rule="(!has(self.status) || !self.status.conditions.exists(c, c.type == 'Programmed' && c.status == 'True')) ? true : (!has(self.spec.awsTransitGateway) ? true : oldSelf.spec.awsTransitGateway.name == self.spec.awsTransitGateway.name)",message="spec.awsTransitGateway.name is immutable when transit gateway is already Programmed"
+// +kubebuilder:validation:XValidation:rule="(!has(self.status) || !self.status.conditions.exists(c, c.type == 'Programmed' && c.status == 'True')) ? true : (!has(self.spec.azureTransitGateway) ? true : oldSelf.spec.azureTransitGateway.name == self.spec.azureTransitGateway.name)",message="spec.azureTransitGateway.name is immutable when transit gateway is already Programmed"
+// +kong:channels=gateway-operator
+type KonnectCloudGatewayTransitGateway struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of KonnectCloudGatewayTransitGateway.
+	Spec KonnectCloudGatewayTransitGatewaySpec `json:"spec"`
+
+	// Status defines the observed state of KonnectCloudGatewayTransitGateway.
+	Status KonnectCloudGatewayTransitGatewayStatus `json:"status,omitempty"`
+}
+
+// KonnectCloudGatewayTransitGatewaySpec defines the desired state of KonnectCloudGatewayTransitGateway.
+//
+// +kubebuilder:validation:XValidation:rule="self.networkRef.type == 'namespacedRef'", message = "only namespacedRef is supported currently"
+type KonnectCloudGatewayTransitGatewaySpec struct {
+	// NetworkRef is the schema for the NetworkRef type.
+	//
+	// +kubebuilder:validation:Required
+	NetworkRef commonv1alpha1.ObjectRef `json:"networkRef"`
+	// KonnectTransitGatewayAPISpec is the configuration of the transit gateway on Konnect side.
+	KonnectTransitGatewayAPISpec `json:",inline"`
+}
+
+// TransitGatewayType defines the type of Konnect transit gateway.
+type TransitGatewayType string
+
+const (
+	// TransitGatewayTypeAWSTransitGateway defines the the AWS transit gateway type.
+	TransitGatewayTypeAWSTransitGateway TransitGatewayType = "awsTransitGateway"
+	// TransitGatewayTypeAzureTransitGateway defines the Azure transit gateway type.
+	TransitGatewayTypeAzureTransitGateway TransitGatewayType = "azureTransitGateway"
+)
```

Comment:
FYI: The API defined consts have changed: https://github.com/Kong/platform-api/pull/1187, https://github.com/Kong/sdk-konnect-go/pull/95/files#diff-c53e1382c018bfbe4bda70043e615edd9de4f4d7aebde916704ca72a7448101eR129-R131.

I _think_ that we still want to define our own `const`s but perhaps we want to reuse the values that are assigned to them given that they now do not contain spaces, WDYT?

Reply:
I prefer to define our own consts. If the API and sdk-konnect-go changes the definition again, importing them may bring breaking changes in upgrading.

Reply:
Fair enough. That makes sense.

If possible then, let's align these values with API consts now. If they diverge in the future then that's another story but let's try to be consistent.

```suggestion
const (
	// TransitGatewayTypeAWSTransitGateway defines the the AWS transit gateway type.
	TransitGatewayTypeAWSTransitGateway TransitGatewayType = "AWSTransitGateway"
	// TransitGatewayTypeAzureTransitGateway defines the Azure transit gateway type.
	TransitGatewayTypeAzureTransitGateway TransitGatewayType = "AzureTransitGateway"
)
```

Reply:
Done.